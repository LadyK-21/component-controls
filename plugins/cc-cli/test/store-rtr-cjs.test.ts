import path from 'path';
import { runTests } from './run-store-tests';

runTests({
  format: 'cjs',
  renderer: 'rtr',
  config: path.resolve(__dirname, '.config'),
});
