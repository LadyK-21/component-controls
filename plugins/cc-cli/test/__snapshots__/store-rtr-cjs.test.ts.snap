// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rtr  cjs 1`] = `
"const path = require('path');
const MatchMediaMock = require('jest-matchmedia-mock').default;
const {
  loadConfigurations,
  extractDocuments,
} = require('@component-controls/config');
const { renderExample } = require('@component-controls/test-renderers');
const { render: reactRender } = require('@component-controls/render/react');
const renderer = require('react-test-renderer');
const act = renderer.act;

const renderErr = () => {
  throw new Error('Could not render the story');
};
const componentErr = () => {
  throw new Error('Error rendering component with react-test-renderer');
};

describe('component-controls generated', () => {
  let matchMedia;
  beforeAll(() => {
    jest.mock('rc-util/lib/Portal');
    matchMedia = new MatchMediaMock();
  });
  afterEach(() => {
    matchMedia.clear();
  });
  const configPath = path.resolve(__dirname, '.config');
  const config = loadConfigurations(configPath);
  if (!config.renderFn) {
    config.renderFn = reactRender;
  }
  const documents = extractDocuments({ config, configPath });
  if (documents) {
    documents.forEach(file => {
      const exports = require(file);
      const doc = exports.default;
      const examples = Object.keys(exports)
        .filter(key => key !== 'default')
        .map(key => exports[key]);

      if (examples.length) {
        describe(doc.title, () => {
          examples.forEach(example => {
            it(example.name, async () => {
              let rendered;
              act(() => {
                rendered = renderExample({
                  example,
                  doc,
                  config,
                });
              });
              if (!rendered) {
                renderErr();
                return;
              }
              const component = renderer.create(rendered);
              if (!component) {
                componentErr();
                return;
              }
              expect(component.toJSON()).toMatchSnapshot();
            });
          });
        });
      }
    });
  }
});
"
`;
